src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAHelloWorldProject.cs

```diff
@@ -702,73 +598,61 @@ public void PublishRelease_does_not_override_Configuration_property_across_forma
             Assert.False(File.Exists(releaseAssetPath)); // build will produce a debug asset, need to make sure this doesn't exist either.
         }
 
-        [Fact]
-        public void PublishRelease_does_not_override_Configuration_option()
+        [Theory]
+        [InlineData("net7.0")]
+        [InlineData("net8.0")]
+        public void It_publishes_with_Release_by_default_in_net_8_but_not_net_7(string tfm)
         {
-            var helloWorldAsset = _testAssetsManager
-               .CopyTestAsset("HelloWorld", "PublishReleaseHelloWorldCsProjConfigOptionOverride")
-               .WithSource()
-               .WithTargetFramework(ToolsetInfo.CurrentTargetFramework)
-               .WithProjectChanges(project =>
-               {
-                   var ns = project.Root.Name.Namespace;
-                   var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
-                   propertyGroup.Add(new XElement(ns + "PublishRelease", "true"));
-               });
-
-            new BuildCommand(helloWorldAsset)
-           .Execute()
-           .Should()
-           .Pass();
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            testProject.RecordProperties("Configuration");
+            testProject.RecordProperties("DebugSymbols"); // If Configuration is set too late, it doesn't actually do anything. Check this too.
 
-            var publishCommand = new DotnetPublishCommand(Log, helloWorldAsset.TestRoot);
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
 
-            publishCommand
-            .Execute("--configuration", "Debug")
-            .Should()
-            .Pass();
+            new DotnetPublishCommand(Log)
+                .WithWorkingDirectory(Path.Combine(testAsset.TestRoot, testProject.Name))
+                .Execute()
+                .Should()
+                .Pass();
 
-            var expectedAssetPath = System.IO.Path.Combine(helloWorldAsset.Path, "bin", "Debug", ToolsetInfo.CurrentTargetFramework, "HelloWorld.dll");
-            Assert.True(File.Exists(expectedAssetPath));
-            var releaseAssetPath = System.IO.Path.Combine(helloWorldAsset.Path, "bin", "Release", ToolsetInfo.CurrentTargetFramework, "HelloWorld.dll");
-            Assert.False(File.Exists(releaseAssetPath)); // build will produce a debug asset, need to make sure this doesn't exist either.
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: tfm, configuration: tfm == "net7.0" ? "Debug" : "Release");
+            var finalConfiguration = properties["Configuration"];
+            var finalDebugSymbols = properties["DebugSymbols"];
+            Assert.True(finalConfiguration == (tfm == "net7.0" ? "Debug" : "Release"));
+            Assert.True(finalDebugSymbols == (tfm == "net7.0" ? "true" : "false"));
```

Comment:
`Assert.Equals` is generally better in situations like this, because if it fails you get better information in the failure message.

```suggestion
            Assert.Equal((tfm == "net7.0" ? "Debug" : "Release"), finalConfiguration);
            Assert.Equal((tfm == "net7.0" ? "true" : "false"), finalDebugSymbols);
```